### ISSUE TITLE
Token authentication with 'legacy' method no longer works

### ISSUE THREAD

**Issue by davidoesch** [abertschi/postcard_creator_wrapper#33]
I tought this was temporary but I can reproduce it now everytime

steps to reproduce

- [X ] OS: Win10 (fresh install, latest updates)
- [X] with latest Python 3.9.5 (tags/v3.9.5:0a7dcbd, May  3 2021, 17:27:52) [MSC v.1928 64 bit (AMD64)] on win32
- [X] same behavior on Linux...
- [X] Network: swisscom and UPC ( so I thought initially it was related to the provider - it is not)
- [X] username password work with the official postcardcreator app (not linked to a swissid, normal account, tried as well other account which work with the official app)
- [X] it works with a swissid account, but not with a standard username / password  Swiss Post Customer Login

started python

>  Python 3.9.5 (tags/v3.9.5:0a7dcbd, May  3 2021, 17:27:52) [MSC v.1928 64 bit (AMD64)] on win32
> Type "help", "copyright", "credits" or "license" for more information.
> from postcard_creator.postcard_creator import PostcardCreator, Postcard, Token, Recipient, Sender
> token = Token()
> token.fetch_token(username='myusername',password='mypassword')

results in

> Traceback (most recent call last):
>   File "c:\temp\venv\lib\site-packages\postcard_creator\token.py", line 276, in _get_access_token_swissid
>     url = resp.json()['nextAction']['successUrl']
> KeyError: 'nextAction'
> 
> During handling of the above exception, another exception occurred:
> 
> Traceback (most recent call last):
>   File "<stdin>", line 1, in <module>
>   File "c:\temp\venv\lib\site-packages\postcard_creator\token.py", line 129, in fetch_token
>     raise e
>   File "c:\temp\venv\lib\site-packages\postcard_creator\token.py", line 123, in fetch_token
>     access_token = self._get_access_token_swissid(session, username, password)
>   File "c:\temp\venv\lib\site-packages\postcard_creator\token.py", line 279, in _get_access_token_swissid
>     raise PostcardCreatorException("failed to login, username/password wrong?")
> postcard_creator.postcard_creator.PostcardCreatorException: failed to login, username/password wrong?

any ideas?

**Comment by abertschi** [comment]
You can verify that the account producing above exception successfully works on the official website? After too many failed login attempts an account usually get blocked and must be unblocked on https://service.post.ch/pccweb/public/ui/view/home via login.

> Traceback (most recent call last):
File "c:\temp\venv\lib\site-packages\postcard_creator\token.py", line 276, in _get_access_token_swissid
url = resp.json()['nextAction']['successUrl']
KeyError: 'nextAction'

> During handling of the above exception, another exception occurred:
> Traceback (most recent call last):
File "", line 1, in
File "c:\temp\venv\lib\site-packages\postcard_creator\token.py", line 129, in fetch_token
raise e
File "c:\temp\venv\lib\site-packages\postcard_creator\token.py", line 123, in fetch_token
access_token = self._get_access_token_swissid(session, username, password)
File "c:\temp\venv\lib\site-packages\postcard_creator\token.py", line 279, in _get_access_token_swissid
raise PostcardCreatorException("failed to login, username/password wrong?")
postcard_creator.postcard_creator.PostcardCreatorException: failed to login, username/password wrong?

The exception does not say much. The Token class probably tried to login with legacy credentials (if method=mixed as default, https://github.com/abertschi/postcard_creator_wrapper/blob/4ca207e254e86aaead0e64286e94d15bc384f939/postcard_creator/token.py#L89) and falls back to swissid if legacy fails. swissid login attempt then fails with your aforementioned message.

**Comment by davidoesch** [comment]
- login with all the used accoutns works perfectly on https://service.post.ch/pccweb/public/ui/view/home 

attached the logging output (trace mode)
[log_pc.txt](https://github.com/abertschi/postcard_creator_wrapper/files/6738364/log_pc.txt)

BTW it stopped working on June 21 --seems like post changed sth

**Comment by abertschi** [comment]
Thanks for shouting out. I verified the issue with a legacy account. You are right, legacy token method does currently not work anymore.

You now have two options:
- upgrade your account to swissid
- wait until the issue is fixed. However, given that many accounts use swissid now, this issue has low priority for me.

**Comment by davidoesch** [comment]
Thx for confirming.
I have several Legacy accounts. Will try to switch to swissid ( but I doubt that I can have more  than one Account  …. For different projects)
Swissid is a little bit cumbersome—- not that widespread in use ( e-I’d discussion) 

thx for all the work you’ve done so far

**Comment by wp99cp** [comment]
The whole thing is connected with the changeover of the Postcard Creator Webpage. This is now hosted on https://service.post.ch/pccweb/public/ui/view/home. See official info mail: 

_Der PostCard Creator zieht um und folgt der neusten Technik. Mit diesem Schritt ändert sich für Sie wenig. Sie finden PostCard Creator weiterhin unter der gewohnten Adresse. Dieser Umzug bedingt jedoch, dass Sie Altes zurücklassen, dafür aber von neuen Gestaltungsmöglichkeiten profitieren werden – und das ist erst der Anfang. Was bedeutet das für Sie? Ihre alten Entwürfe und Mailings können wir nicht mitnehmen und Sie haben ab dem 22. Juni 2021 keinen Zugriff mehr auf Ihre bisher gestalteten Postkarten._

At the same time, the old API was shut down. As was already the case for the SwissID Sign in, we have to adapt the API calls to the new API; this should not be a problem, since we can use the same calls as for the swissID sign-in. The only "new thing" / "bug to fix" is to change the sign-in process in order to obtain a valid token for the new API with the legacy sign-in method.

**Comment by wp99cp** [comment]
I've tried to implement the new sign-in process. It works analogously to the SwissID sign-in. Except it used swiss posts internal IDP instead of the external SwissID IDP. But in the end, I get an error: ```Access denied by resource owner or authorization server```. 

@abertschi you have implemented the SwissID sign-in process, right? Did you encounter a similar error during your implementation? And do you know how to fix this error? It's probably some cookies that are missing.

Link to my code (reproducing the error): https://gist.github.com/wp99cp/2b1ef687a39f47c495e34a5e5bde764f

**Comment by abertschi** [comment]
#34
